src/v/cluster/tm_stm.cc

```diff
@@ -517,34 +533,36 @@ ss::future<tm_stm::op_status> tm_stm::add_partitions(
         if (!r.has_value()) {
             co_return tm_stm::op_status::unknown;
         }
-        _mem_txes[tx_id] = tx;
         co_return tm_stm::op_status::success;
     }
 
+    auto tx_it = do_get_tx_it(tx_id);
+    if (!tx_it) {
+        // Unlikely to happen.
+        co_return tm_stm::op_status::unknown;
+    }
     for (auto& partition : partitions) {
-        ptx->second.partitions.push_back(partition);
+        tx_it.value()->second.partitions.push_back(partition);
```

Comment:
please reactor this code, current version may accidentally modify `log_state`

Reply:
What do you suggest refactoring to, copying it over to mem_txes? I thought this unfortunate mixing up of log/mem states is acceptable and we can change any where.